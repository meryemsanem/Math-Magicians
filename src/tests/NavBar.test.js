import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

test('renders the Navbar component', () => {
  render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>,
  );

  const headerText = screen.getByText(/Math Magicians/i);
  expect(headerText).toBeInTheDocument();

  const homeLink = screen.getByRole('link', { name: /Home/i });
  expect(homeLink).toBeInTheDocument();

  const calculatorLink = screen.getByRole('link', { name: /Calculator/i });
  expect(calculatorLink).toBeInTheDocument();

  const quoteLink = screen.getByRole('link', { name: /Quote/i });
  expect(quoteLink).toBeInTheDocument();

  const navbarComponent = screen.getByRole('navigation');
  expect(navbarComponent).toMatchSnapshot();
});
