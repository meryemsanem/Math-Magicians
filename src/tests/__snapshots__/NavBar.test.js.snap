// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Navbar component 1`] = `
<nav>
  <h1>
    Math Magicians
  </h1>
  <ul>
    <li>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        class=""
        href="/calculator"
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        class=""
        href="/quote"
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
